### PR TITLE
[Dependency Scanning] Record whether discovered binary Swift modules are frameworks

### PR DESCRIPTION
**Pull request by artemcm**
This info is required so that it can be specified as input to compilation that uses such module as a dependency, in order to decide whether or not to insert an auto-linking directive for it.

Part of rdar://102824777

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
https://github.com/apple/swift-driver/pull/1240
@swift-ci test

File: lib/Frontend/ModuleInterfaceLoader.cpp [hunk 1]
```diff
@@ -1993,6 +1993,24 @@ bool ExplicitSwiftModuleLoader::canImportModule(
...
+  if (!moduleBuf) {
+    Ctx.Diags.diagnose(SourceLoc(), diag::error_opening_explicit_module_file,
+                       it->second.modulePath);
+    return false;
+  }
+  auto metaData = serialization::validateSerializedAST(
+      (*moduleBuf)->getBuffer(), Ctx.SILOpts.EnableOSSAModules,
+      Ctx.LangOpts.SDKName, !Ctx.LangOpts.DebuggerSupport);
+  versionInfo->setVersion(metaData.userModuleVersion,
+                          ModuleVersionSourceKind::SwiftBinaryModule);
```
Comment (nkcsgexi) [inline lib/Frontend/ModuleInterfaceLoader.cpp:2013]: This change seems to be from the other PR.

**Review by nkcsgexi — APPROVED** [review]
LGTM

**Comment by artemcm** [comment]
https://github.com/apple/swift-driver/pull/1240
@swift-ci test

File: lib/Frontend/ModuleInterfaceLoader.cpp (hunk 1 above)
Comment (artemcm) [reply lib/Frontend/ModuleInterfaceLoader.cpp]: Re-based on `main`, thanks for the catch.

**Review by owenv — APPROVED** [review]

**Pull request merged**